esp/src/src-react/components/Title.tsx

```diff
@@ -62,18 +76,46 @@ export const DevTitle: React.FunctionComponent<DevTitleProps> = ({
                     }
                 },
                 { key: "divider_3", itemType: ContextualMenuItemType.Divider },
-                { key: "lock", text: nlsHPCC.Lock },
-                { key: "logout", text: nlsHPCC.Logout },
+                {
+                    key: "lock", text: nlsHPCC.Lock, disabled: !currentUser?.username, onClick: () => {
+                        fetch("esp/lock", {
+                            method: "post"
+                        }).then(() => { window.location.href = "/esp/files/Login.html"; });
+                    }
+                },
+                {
+                    key: "logout", text: nlsHPCC.Logout, disabled: !currentUser?.username, onClick: () => {
+                        fetch("esp/logout", {
+                            method: "post"
+                        }).then(data => {
+                            if (data) {
+                                cookie("ECLWatchUser", "", { expires: -1 });
+                                cookie("ESPSessionID" + location.port + " = '' ", "", { expires: -1 });
+                                cookie("Status", "", { expires: -1 });
+                                cookie("User", "", { expires: -1 });
+                                window.location.reload();
+                            }
```

Comment:
@jeclrsg can you open a Jira to add a cookie warning / approval?

Reply:
Issue opened (HPCC-26627)